### PR TITLE
test(agents): add agent MCP smoke lane

### LINKED ISSUE #3284: test(agents): Agent / MCP server 境界の smoke tests を追加する
> Language / 言語: 日本語 | English

Parent roadmap: #3279
Recommended after: #3280 and can proceed in parallel with #3282/#3283 if file ownership is separated

# test(agents): Agent / MCP server 境界の smoke tests を追加する

## 背景

`src/agents` と `src/mcp-server` は ae-framework の中核境界だが、planning audit のヒューリスティックでは直接テストが薄い可能性がある。外部 API や network に依存しない形で、agent input/output と MCP stdio 境界を smoke test として固定したい。

## 目的

Agent / MCP server の最小入出力契約を固定し、provider env safety と JSON-RPC / stdio 境界の regression を検出しやすくする。

## 対象ファイル候補

- `src/agents/*`
- `src/mcp-server/*`
- `src/providers/*`
- `tests/unit/agents/*`
- `tests/codex/*`
- `docs/integrations/*`
- `docs/agents/*`

## TODO

- [ ] `intent-agent` の minimal smoke test を追加する
- [ ] `formal-agent` の minimal smoke test を追加する
- [ ] `verify-agent` の minimal smoke test を追加する
- [ ] agent input / output artifact shape を固定する
- [ ] MCP stdio server の JSON-RPC initialize / tool list / basic call smoke test を追加する
- [ ] provider は network を呼ばない mock / echo provider を中心にテストする
- [ ] OpenAI / Anthropic / Gemini provider の env 未設定時の安全動作をテストする
- [ ] invalid env / missing key / dry-run mode の挙動を固定する
- [ ] `docs/integrations/*` の代表コマンドとテストの対応表を必要に応じて更新する
- [ ] `docs/agents/*` に agent 境界テストの位置づけを追記する

## 受け入れ条件

- 主要 agent の minimal input/output が smoke test で確認できる
- MCP stdio 境界の JSON-RPC 基本動作がテストされている
- provider が env 未設定時に外部 network を呼ばず安全に失敗または dry-run できる
- docs が agent / MCP テストの実行方法を説明している

## 検証コマンド

```bash
pnpm -s exec vitest run tests/unit/agents tests/codex --reporter dot
pnpm -s run types:check
pnpm -s run check:doc-consistency
```

必要に応じて:

```bash
pnpm -s run codex:quickstart
```

## Out of scope

- 実 provider を使った外部 API E2E
- MCP server の長時間 soak test
- full agent pipeline の重い E2E required 化

### PR DESCRIPTION
**Pull request by ootakazuhiko**
## Summary
- Add a focused `test:agents:smoke` Vitest lane for Agent / MCP boundary regression coverage.
- Cover minimal `IntentAgent`, `FormalAgent`, and `VerifyAgent` input/output contracts without external services.
- Exercise the Intent MCP stdio server with an MCP client initialize handshake, tool listing, and basic tool call, and document the lane in testing and CodeX integration docs.

## Acceptance
- Resolves #3284
- Part of #3279
- Agent / MCP boundary smoke coverage is runnable from the repository root.
- Provider API keys unset path falls back to the local echo provider without network access.
- External LLM provider E2E, long-running MCP soak tests, and full agent-pipeline E2E remain out of scope for this slice.

## Verification
- `pnpm -s run test:agents:smoke -- --reporter dot` — 1 file / 3 tests passed.
- `pnpm -s exec vitest run tests/agents tests/unit/agents tests/codex tests/providers/provider-utils.test.ts tests/commands/mcp-command.test.ts --reporter dot` — 23 files / 150 tests passed.
- `pnpm -s run types:check` — passed.
- `pnpm -s run check:doc-consistency` — passed; doc governance warnings 0.
- `pnpm -s run check:doc-todo-markers` — passed.
- `git diff --check` — passed.
- `pnpm -s run lint -- tests/agents/agent-mcp-boundary-smoke.test.ts` — exited 0; command linted the repo with existing warnings and reported the new test file is ignored by the current lint config.

## Rollback
- Revert this PR to remove the focused agent/MCP smoke test lane, the smoke test file, and the documentation route updates. No production runtime behavior or schema contract is changed.

### CODE REVIEW

File: tests/agents/agent-mcp-boundary-smoke.test.ts [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+describe('Agent / MCP boundary smoke', () => {
+  test('intent, formal, and verify agents expose stable minimal input/output shapes', async () => {
+    const intentAgent = new IntentAgent();
+    const intentRequest = IntentAgent.createSimpleRequest(
+      'Users must reserve inventory without making on-hand stock negative.',
+      {
+        domain: 'inventory',
+        analysisDepth: 'basic',
+        outputFormat: 'json',
```
[the remark this answers is not in the document]
Comment (ootakazuhiko) [reply tests/agents/agent-mcp-boundary-smoke.test.ts]: 対応しました。`IntentAgent.createSimpleRequest` の `outputFormat` を契約外の `json` から、宣言済みの有効値である `structured` に変更しました。
